### PR TITLE
feat: expose monadic quote errors to the user

### PR DESCRIPTION
**Pull request by 0xApotheosis**
## Description

We implemented monadic errors for trade quotes, but didn't finish propagating them to the user.

This PR implements that (for a limited set of error codes - we can expand these cases at Product's discretion).

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

N/A

## Risk

Medium regression risk to the display of available swappers.

## Testing

Get a set of trade quotes, and see that we now show swappers (though greyed out and unselectable) when their quotes have the following error codes:

1. `SwapErrorType.TRADING_HALTED`
2. `SwapErrorType.UNSUPPORTED_PAIR`

### Engineering

☝️

### Operations

☝️

## Screenshots (if applicable)

Screenshot 2023-08-09 at 11 45 46 am
Screenshot 2023-08-09 at 11 58 39 am

### CODE REVIEW

**Comment by 0xApotheosis** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #5055** <a href="https://app.graphite.dev/github/pr/shapeshift/web/5055" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/5055?utm_source=stack-comment).

File: src/components/MultiHopTrade/components/TradeInput/TradeInput.tsx [hunk 1]
```diff
@@ -209,8 +209,14 @@ export const TradeInput = memo(() => {
...
 
   const shouldDisablePreviewButton = useMemo(() => {
-    return quoteHasError || manualReceiveAddressIsValidating || isLoading || !isSellAmountEntered
-  }, [isLoading, isSellAmountEntered, manualReceiveAddressIsValidating, quoteHasError])
+    return (
+      quoteHasError ||
+      manualReceiveAddressIsValidating ||
+      isLoading ||
+      !isSellAmountEntered ||
+      !activeQuote
```
Comment (woodenfurniture) [inline src/components/MultiHopTrade/components/TradeInput/TradeInput.tsx:217]: 🍻

File: src/lib/swapper/swappers/ThorchainSwapper/utils/getQuote/getQuote.ts [hunk 2]
```diff
@@ -80,6 +80,14 @@ export const getQuote = async ({
         details: { sellAssetId: sellAsset.assetId, buyAssetId },
       }),
     )
+  } else if (isError && /trading is halted/.test(data.error)) {
+    return Err(
+      makeSwapErrorRight({
+        message: `[getTradeRate]: Trading is halted, cannot process swap`,
+        code: SwapErrorType.TRADING_HALTED,
```
Comment (woodenfurniture) [inline src/lib/swapper/swappers/ThorchainSwapper/utils/getQuote/getQuote.ts:87]: 🧠 this is what we want!! :D

File: src/components/MultiHopTrade/components/TradeInput/components/TradeQuotes/TradeQuote.tsx [hunk 3]
```diff
@@ -160,23 +173,31 @@ export const TradeQuoteLoaded: React.FC<TradeQuoteLoadedProps> = ({
     [activeSwapperColor, focusColor, isActive],
   )
 
-  return totalReceiveAmountCryptoPrecision ? (
+  const isDisabled = !!error
+
+  // TODO: work out for which error codes we want to show a swapper with a human-readable error vs hiding it
```
Comment (woodenfurniture) [inline src/components/MultiHopTrade/components/TradeInput/components/TradeQuotes/TradeQuote.tsx:178]: i reckon we should remove any error codes we dont want to display (as there is no point having them), then render ALL of the ones in the remaining list

**Review by woodenfurniture — APPROVED** [review]
man nice job, this is a step forwards for sure! 🦊

File: src/components/MultiHopTrade/components/TradeInput/components/TradeQuotes/TradeQuote.tsx (hunk 3 above)
Comment (0xApotheosis) [reply src/components/MultiHopTrade/components/TradeInput/components/TradeQuotes/TradeQuote.tsx:178]: Big 🧠

**Pull request merged**